# 🍫 Chocolate Sales Analysis using Pandas

### Data Analysis Project

**Tools Used:**
- Python
- Pandas

## 📌 Project Overview

This project analyzes chocolate sales data using the Pandas library.

### Objectives
- Clean the dataset
- Explore sales trends
- Analyze product performance
- Identify top-performing salespersons
- Generate business insights

## 📂 Import Libraries

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

## 📁 Load Dataset

In [122]:
df = pd.read_csv("Chocolate_Sales.csv")
df.head()

,Order_ID,Product,Country,Channel,Salesperson,Order_Date,Discount_Pct,Price_per_Box,Marketing_Spend,Boxes_Shipped,Amount
0,ORD-069833,Truffle Gift Box,Australia,Retail,Arjun Mehta,2022-12-11,3.5,13.72,202.03,71,912.31
1,ORD-090726,85% Dark Bar,Australia,Retail,Arjun Mehta,2023-03-14,9.4,3.30,55.18,84,245.91
2,ORD-042159,70% Dark Bar,Japan,Retail,Hannah Müller,2023-12-21,4.9,18.21,60.65,35,583.7
3,ORD-197166,Hazelnut Milk Bar,Germany,Retail,Arjun Mehta,2023-12-18,15.0,2.66,52.00,92,211.27
4,ORD-112162,Almond Crunch Bar,Australia,Retail,Yuki Sato,2023-08-18,4.4,2.75,187.44,214,549.69


## 🔍 Initial Data Inspection

In this section, we examine the dataset's structure, columns, and missing values.

In [123]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200000 entries, 0 to 199999
Data columns (total 11 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   Order_ID         200000 non-null  object 
 1   Product          200000 non-null  object 
 2   Country          200000 non-null  object 
 3   Channel          200000 non-null  object 
 4   Salesperson      200000 non-null  object 
 5   Order_Date       199563 non-null  object 
 6   Discount_Pct     199511 non-null  float64
 7   Price_per_Box    199543 non-null  float64
 8   Marketing_Spend  199539 non-null  float64
 9   Boxes_Shipped    200000 non-null  int64  
 10  Amount           200000 non-null  object 
dtypes: float64(3), int64(1), object(7)
memory usage: 16.8+ MB


In [124]:
df.describe()

,Discount_Pct,Price_per_Box,Marketing_Spend,Boxes_Shipped
count,199511.000000,199543.000000,199539.000000,200000.000000
mean,13.107457,5.911507,95.450141,139.584105
std,6.374477,4.954976,66.243760,118.665393
min,0.000000,2.040000,4.300000,-1642.000000
25%,8.300000,2.950000,52.270000,67.000000
50%,12.600000,3.280000,78.850000,113.000000
75%,17.500000,4.570000,118.710000,180.000000
max,38.300000,21.650000,798.222443,3811.000000


In [125]:
df.sample()

,Order_ID,Product,Country,Channel,Salesperson,Order_Date,Discount_Pct,Price_per_Box,Marketing_Spend,Boxes_Shipped,Amount
100196,ORD-190176,Mixed Assortment Box,Australia,Retail,James Carter,2022-12-28,10.0,3.5,41.37,200,597.12


In [126]:
df.tail()

,Order_ID,Product,Country,Channel,Salesperson,Order_Date,Discount_Pct,Price_per_Box,Marketing_Spend,Boxes_Shipped,Amount
199995,ORD-053460,70% Dark Bar,Australia,Retail,Beatriz Souza,2022-05-27,10.6,3.39,186.19,125,418.56
199996,ORD-010743,Truffle Gift Box,India,Wholesale,Arjun Mehta,2022-09-06,17.3,3.07,42.28,148,366.47
199997,ORD-049690,Salted Caramel Bar,Japan,Online,Arjun Mehta,2022-06-16,18.3,2.74,100.19,149,329.79
199998,ORD-189637,85% Dark Bar,Japan,Wholesale,Rohan Patel,2022-10-14,9.3,14.30,29.96,24,304.51
199999,ORD-061616,Mint Dark Bar,Australia,Retail,Hannah Müller,2022-01-25,7.2,3.01,270.86,131,366.91


In [127]:
df.shape

(200000, 11)

In [128]:
df.columns

Index(['Order_ID', 'Product', 'Country', 'Channel', 'Salesperson',
       'Order_Date', 'Discount_Pct', 'Price_per_Box', 'Marketing_Spend',
       'Boxes_Shipped', 'Amount'],
      dtype='object')

In [129]:
df.dtypes

Order_ID            object
Product             object
Country             object
Channel             object
Salesperson         object
Order_Date          object
Discount_Pct       float64
Price_per_Box      float64
Marketing_Spend    float64
Boxes_Shipped        int64
Amount              object
dtype: object

In [130]:
df.isnull().sum()

Order_ID             0
Product              0
Country              0
Channel              0
Salesperson          0
Order_Date         437
Discount_Pct       489
Price_per_Box      457
Marketing_Spend    461
Boxes_Shipped        0
Amount               0
dtype: int64

In [131]:
df.duplicated().sum()

np.int64(0)

## 🧹 Data Cleaning

The following cleaning steps are performed:
- Removed duplicate records
- Handled missing values
- Corrected data types
- Renamed columns where necessary

In [132]:
df.columns

Index(['Order_ID', 'Product', 'Country', 'Channel', 'Salesperson',
       'Order_Date', 'Discount_Pct', 'Price_per_Box', 'Marketing_Spend',
       'Boxes_Shipped', 'Amount'],
      dtype='object')

In [133]:
df.columns = df.columns.str.lower()
df.rename(columns={'amount': 'revenue'}, inplace=True)

In [134]:
df.columns

Index(['order_id', 'product', 'country', 'channel', 'salesperson',
       'order_date', 'discount_pct', 'price_per_box', 'marketing_spend',
       'boxes_shipped', 'revenue'],
      dtype='object')

In [135]:
df['revenue'] = df['revenue'].str.replace('$', '', regex=False)

In [136]:
df['revenue'] = df['revenue'].astype('float')

## Missing Value Treatment

### Missing Values Identified

| Column | Missing Values | Treatment |
|----------|----------|----------|
| Order_Date | 437 | Rows removed using `dropna()` |
| Discount_Pct | 489 | Missing values filled with `0` |
| Price_per_Box | 457 | Calculated using `Amount / Boxes_Shipped` |
| Marketing_Spend | 461 | Missing values filled with `0` |

### Reasoning

- `Order_Date` contained 437 missing values out of 200,000 records (0.22% of the dataset). Since the percentage was very small, the affected rows were removed.
- Missing values in `Discount_Pct` were replaced with `0`, assuming no discount was applied.
- Missing values in `Marketing_Spend` were replaced with `0`, assuming no marketing spend was recorded.
- Missing values in `Price_per_Box` were derived using the business rule:

  `Price_per_Box = Amount / Boxes_Shipped`

  This allowed missing values to be reconstructed rather than using an average or median value.

In [137]:
df['price_per_box'] = df['price_per_box'].fillna(
    df['revenue'] / df['boxes_shipped']
)

In [138]:
df['discount_pct'] = df['discount_pct'].fillna(0)

In [139]:
df['marketing_spend'] = df['marketing_spend'].fillna(0)

In [140]:
df = df.dropna(subset=['order_date'])

In [141]:
df.isnull().sum()

order_id           0
product            0
country            0
channel            0
salesperson        0
order_date         0
discount_pct       0
price_per_box      0
marketing_spend    0
boxes_shipped      0
revenue            0
dtype: int64

In [142]:
df['order_date'] = pd.to_datetime(
    df['order_date'],
    format='mixed',
    dayfirst=True,
    errors='coerce'
)

In [143]:
df['order_date'].dtype

dtype('<M8[ns]')

In [144]:
df['salesperson'] = df['salesperson'].str.strip()

In [145]:
df['country'] = df['country'].str.title()

In [146]:
df.columns

Index(['order_id', 'product', 'country', 'channel', 'salesperson',
       'order_date', 'discount_pct', 'price_per_box', 'marketing_spend',
       'boxes_shipped', 'revenue'],
      dtype='object')

## Date-Based Columns

To support time-series analysis and dashboard development, additional date-related columns were derived from the `order_date` field after converting it to the `datetime64[ns]` data type.

### New Columns Created

| Column Name | Description |
|------------|-------------|
| `year` | Extracted year from the order date |
| `month_num` | Month number (1–12) used for chronological sorting |
| `month_name` | Month name used for reporting and visualizations |
| `weekday_num` | Weekday number (0–6) used for sorting |
| `weekday_name` | Day name (Monday–Sunday) used for trend analysis |
| `quarter` | Quarter of the year (Q1–Q4) |

### Purpose

These columns were created to:

- Analyze revenue trends by month and year.
- Compare sales performance across weekdays.
- Enable proper sorting of months and weekdays in visualizations.
- Support quarterly business performance analysis.
- Improve dashboard interactivity through date-based filtering and slicing.


In [147]:
df['year'] = df['order_date'].dt.year

In [148]:
df['month_num'] = df['order_date'].dt.month

In [149]:
df['month_name'] = df['order_date'].dt.month_name()

In [150]:
df['weekday_num'] = df['order_date'].dt.weekday

In [151]:
df['weekday_name'] = df['order_date'].dt.day_name()

In [152]:
df['quarter'] = df['order_date'].dt.quarter

In [153]:
#KPis
#Total-Orders
df['order_id'].nunique()

199563

In [154]:
#Total-Revenue
df['revenue'].sum()

np.float64(102337823.64)

In [155]:
#Average-Discount %
round(df['discount_pct'].mean(),2)

np.float64(13.08)

In [156]:
#Total-Boxes-Shipped

df['boxes_shipped'].sum()

np.int64(27852110)

In [157]:
#Monthly-Revenue-Trend
monthly_revenue = (
    df.groupby(['year','month_num','month_name'])['revenue']
      .sum()
      .reset_index()
)

monthly_revenue

,year,month_num,month_name,revenue
0,2022,1,January,4653016.49
1,2022,2,February,4012464.65
2,2022,3,March,3951859.39
3,2022,4,April,3785674.18
4,2022,5,May,3875720.61
5,2022,6,June,3581998.51
6,2022,7,July,3712123.08
7,2022,8,August,3945404.35
8,2022,9,September,3977491.83
9,2022,10,October,4253114.80


In [158]:
#Weekday-Revenue

weekday_revenue = (
    df.groupby(['weekday_num','weekday_name'])['revenue']
      .sum()
      .reset_index()
      .sort_values('weekday_num')
)

weekday_revenue

,weekday_num,weekday_name,revenue
0,0,Monday,14432995.81
1,1,Tuesday,14552399.93
2,2,Wednesday,14565468.60
3,3,Thursday,14649986.24
4,4,Friday,14547946.37
5,5,Saturday,14759611.13
6,6,Sunday,14829415.56


In [159]:
#Revenue-By-Country

country_revenue = (
    df.groupby('country')['revenue']
      .sum()
      .sort_values(ascending=False)
)

country_revenue

country
Australia    42332377.82
Brazil       36545432.34
Germany      11333175.86
India         6478907.41
Japan         5647930.21
Name: revenue, dtype: float64

In [160]:
#Revenue-By-Channel


channel_revenue = df.groupby('channel')['revenue'].sum().sort_values(ascending = False)

channel_revenue

channel
Retail       53662210.59
Wholesale    37407371.97
Online       11268241.08
Name: revenue, dtype: float64

In [161]:
#Boxes-shipped-by-channel

channel_boxes = (df.groupby('channel')['boxes_shipped']
               .sum()
               .sort_values(ascending=False))
channel_boxes

channel
Retail       14606514
Wholesale    10178798
Online        3066798
Name: boxes_shipped, dtype: int64

In [162]:
#Top-products-by-revenue

product_revenue = (
    df.groupby('product')['revenue']
      .sum()
      .sort_values(ascending=False)
)

product_revenue.head(10)

product
70% Dark Bar            26311208.87
Mixed Assortment Box    19485568.10
Truffle Gift Box        19358724.31
Milk Classic Bar         6237124.16
Almond Crunch Bar        5773121.16
Pralines Gift Box        5751633.64
Choco Coins Bag          4769798.21
Mint Dark Bar            3508764.60
White Choco Bar          3218809.33
85% Dark Bar             2962750.27
Name: revenue, dtype: float64

In [163]:
#Top-sales-persons

salesperson_revenue = (
    df.groupby('salesperson')['revenue']
      .sum()
      .sort_values(ascending=False)
)

salesperson_revenue.head(10)

salesperson
Arjun Mehta        34616018.59
Priya Sharma       14991455.49
Emily Clarke        9101838.36
Lucas Walker        6362312.42
Rohan Patel         4926871.27
Yuki Sato           4064463.21
James Carter        3314261.08
Rafael Oliveira     2824017.86
Felix Becker        2448016.19
Haruto Tanaka       2204466.76
Name: revenue, dtype: float64

In [164]:
#relation bw major columns

df[['revenue',
    'discount_pct',
    'marketing_spend',
    'boxes_shipped']].corr()

,revenue,discount_pct,marketing_spend,boxes_shipped
revenue,1.000000,0.063583,0.371251,0.580050
discount_pct,0.063583,1.000000,0.040532,0.168636
marketing_spend,0.371251,0.040532,1.000000,0.316061
boxes_shipped,0.580050,0.168636,0.316061,1.000000


## Executive Summary & Key Business Insights

This analysis was conducted on a sales dataset containing **199,563 orders** after data cleaning and preprocessing. The objective was to evaluate sales performance across products, countries, channels, sales representatives, and time periods while identifying relationships between key business metrics.

### KPI Summary

| KPI | Value |
|------|------:|
| Total Orders | 199,563 |
| Total Revenue | $102.34 Million |
| Average Discount | 13.08% |
| Total Boxes Shipped | 27.85 Million |

---

## Revenue Trend Analysis

### Monthly Revenue Trends

- Revenue exhibited a consistent seasonal pattern across both years.
- **December** generated the highest revenue in both 2022 and 2023, indicating strong year-end demand.
- **June** recorded the lowest revenue period across both years.
- Revenue generally increased during the fourth quarter (Q4), suggesting seasonal purchasing behavior.

### Weekday Performance

- **Sunday** generated the highest revenue ($14.83M).
- **Saturday** was the second-best performing day ($14.76M).
- Revenue distribution remained relatively balanced throughout the week.
- Weekend sales slightly outperformed weekday sales, indicating stronger customer purchasing activity during weekends.

---

## Geographic Performance Analysis

### Revenue by Country

| Country | Revenue |
|----------|---------:|
| Australia | $42.33M |
| Brazil | $36.55M |
| Germany | $11.33M |
| India | $6.48M |
| Japan | $5.65M |

### Key Findings

- Australia was the largest revenue-generating market, contributing over 41% of total revenue.
- Brazil was the second strongest market.
- India and Japan contributed comparatively lower revenue and may represent growth opportunities.

---

## Channel Performance Analysis

### Revenue by Channel

| Channel | Revenue |
|----------|---------:|
| Retail | $53.66M |
| Wholesale | $37.41M |
| Online | $11.27M |

### Boxes Shipped by Channel

| Channel | Boxes Shipped |
|----------|--------------:|
| Retail | 14.61M |
| Wholesale | 10.18M |
| Online | 3.07M |

### Key Findings

- Retail was the highest-performing channel in both revenue and shipment volume.
- Wholesale contributed significantly to overall sales performance.
- Online sales represented the smallest channel and may offer opportunities for digital growth initiatives.

---

## Product Performance Analysis

### Top Revenue-Generating Products

| Product | Revenue |
|----------|---------:|
| 70% Dark Bar | $26.31M |
| Mixed Assortment Box | $19.49M |
| Truffle Gift Box | $19.36M |
| Milk Classic Bar | $6.24M |
| Almond Crunch Bar | $5.77M |

### Key Findings

- The **70% Dark Bar** was the highest-performing product.
- Premium gift products such as **Mixed Assortment Box** and **Truffle Gift Box** generated substantial revenue.
- Revenue is concentrated among a small number of top-performing products.

---

## Salesperson Performance Analysis

### Top Revenue-Generating Salespersons

| Salesperson | Revenue |
|-------------|---------:|
| Arjun Mehta | $34.62M |
| Priya Sharma | $14.99M |
| Emily Clarke | $9.10M |
| Lucas Walker | $6.36M |
| Rohan Patel | $4.93M |

### Key Findings

- Arjun Mehta significantly outperformed all other sales representatives.
- Revenue generation is concentrated among a small group of high-performing salespersons.
- Top performers contributed a substantial share of total company revenue.

---

## Correlation Analysis

Correlation analysis was performed to understand relationships between key business variables.

| Relationship | Correlation |
|-------------|------------:|
| Revenue ↔ Boxes Shipped | 0.58 |
| Revenue ↔ Marketing Spend | 0.37 |
| Marketing Spend ↔ Boxes Shipped | 0.32 |
| Discount % ↔ Boxes Shipped | 0.17 |
| Revenue ↔ Discount % | 0.06 |

### Key Findings

- Revenue has the strongest positive relationship with shipment volume (**0.58**).
- Marketing spend shows a moderate positive relationship with revenue (**0.37**).
- Marketing spend also positively influences shipment volume (**0.32**).
- Discount percentage has minimal impact on revenue (**0.06**).
- Discounts show only a weak relationship with shipment volume (**0.17**).

---

## Business Recommendations

1. Focus on increasing shipment volume, as it is the strongest driver of revenue.
2. Continue investing in marketing activities that contribute positively to sales performance.
3. Prioritize high-performing products such as the 70% Dark Bar and premium gift boxes.
4. Analyze the strategies used by top-performing salespersons and replicate best practices across the sales team.
5. Explore opportunities to strengthen online sales channels and expand lower-performing geographic markets.

---

## Next Steps

The cleaned dataset and analytical findings will be utilized to develop an interactive dashboard in a visualization tool such as **Power BI**. The dashboard will provide stakeholders with dynamic insights into:

- Sales performance trends
- Geographic revenue distribution
- Channel effectiveness
- Product performance
- Salesperson performance
- Marketing and discount impact analysis

This dashboard will support data-driven decision-making and enable ongoing business performance monitoring.

In [167]:
df.to_excel('chocolate_sales_cleaned.xlsx', index=False)